core/src/main/scala/com/avast/clients/storage/StorageClient.scala

```diff
@@ -0,0 +1,21 @@
+package com.avast.clients.storage
+
+import better.files.File
+import cats.effect.Effect
+import com.avast.scala.hashes.Sha256
+
+import scala.language.higherKinds
+
+trait StorageClient[F[_]] extends AutoCloseable {
+  def head(sha256: Sha256): F[Either[StorageException, HeadResult]]
+
+  def get(sha256: Sha256, dest: File = File.newTemporaryFile(prefix = "stor")): F[Either[StorageException, GetResult]]
+}
+
+class DefaultStorageClient[F[_]: Effect](backend: StorageBackend[F]) extends StorageClient[F] {
```

Comment:
Maybe `SingleStorageClient`?

Reply:
That would be misleading name. The client uses single `StorageBackend` instance however it can be in fact _merged_ from two (or more) instances.